### PR TITLE
Run NativeAot tests on arm64

### PR DESCRIPTION
**Pull request by sbomer**

### CODE REVIEW

File: src/tests/nativeaot/CustomMain/CustomMain.csproj [hunk 1]
```diff
@@ -3,6 +3,7 @@
     <OutputType>Exe</OutputType>
     <CustomNativeMain>true</CustomNativeMain>
     <StaticLibraryPrefix Condition="'$(TargetOS)' != 'windows'">lib</StaticLibraryPrefix>
+    <NativeAotIncompatible Condition="'$(TargetArchitecture)' == 'arm64' and '$(TargetOS)' != 'windows'">true</NativeAotIncompatible>
```
Comment (filipnavara) [inline src/tests/nativeaot/CustomMain/CustomMain.csproj:6]: I was pretty sure the test was supposed to check an osx-arm64 scenario so it feels wrong to disable that.

Comment (filipnavara) [reply src/tests/nativeaot/CustomMain/CustomMain.csproj:7]: I was fighting with cross-building the dependencies when trying to run something on maccatalyst-arm64/x64 but it definitely was not broken for the osx-* combinations that I tried.

At minimum, can the condition be relaxed to allow running the test when the scenario is not cross-compilation?

File: src/tests/nativeaot/SmokeTests/ControlFlowGuard/ControlFlowGuard.csproj [hunk 2]
```diff
@@ -6,7 +6,7 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <ControlFlowGuard>Guard</ControlFlowGuard>
     <Optimize>true</Optimize>
-    <CLRTestTargetUnsupported Condition="'$(TargetsWindows)' != 'true'">true</CLRTestTargetUnsupported>
+    <CLRTestTargetUnsupported Condition="'$(TargetsWindows)' != 'true' or '$(TargetArchitecture)' != 'x64'">true</CLRTestTargetUnsupported>
```
Comment (MichalStrehovsky) [inline src/tests/nativeaot/SmokeTests/ControlFlowGuard/ControlFlowGuard.csproj:9]: Could you file an issue on this and link it here? I pulled down the crash dump and this is failing on invalid instruction when running code generated here:

https://github.com/dotnet/runtime/blob/ce719aa21f73ff387956b2f2ab148d32a9adaed7/src/tests/nativeaot/SmokeTests/ControlFlowGuard/ControlFlowGuard.cs#L167-L200

I don't immediately see what's wrong with it. The instruction disassembles into `ret` fine in the crash dump. The test is crashing before we enable CFG enforcement. Maybe I'm missing some additional Windows ARM64 knowledge. I don't quickly know a way to fix this test (maybe @jkotas would know what's wrong with it), so let's track it with an issue. CFG should work.

Comment (jkotas) [reply src/tests/nativeaot/SmokeTests/ControlFlowGuard/ControlFlowGuard.csproj:9]: One obvious problem is missing `FlushInstructionCache` call.

Comment (MichalStrehovsky) [reply src/tests/nativeaot/SmokeTests/ControlFlowGuard/ControlFlowGuard.csproj:9]: > One obvious problem is missing `FlushInstructionCache` call.

Thanks! Likely another couple of hours of troubleshooting saved! :)

**Review by MichalStrehovsky — APPROVED** [review]
Looks great thanks!

But looks like the CI didn't actually run the runtime pipeline for some reason?

**Pull request merged**